### PR TITLE
Make one workflow able to build 5 platforms with reuse workflow

### PR DESCRIPTION
**Pull request by cynthiajoan**
[replace this line]: # (Describe your changes in detail.)
***

[replace this line]: # (Describe your testing in detail.)
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [ ] Other, such as a build process or documentation change.
***

### CODE REVIEW

File: .github/workflows/build_android.yml [hunk 1]
```diff
@@ -8,30 +8,36 @@ on:
...
       additional_cmake_flags:
         description: 'Additional flags to pass into CMake'
         default: ''
         required: false
+        type: string
 
 jobs:
   build_android:
-    name: build-android-macos-unity${{ github.event.inputs.unity_version }}-CPP${{ github.event.inputs.firebase_cpp_sdk_version }}
+    name: build-android-macos-unity${{ inputs.unity_version }}-CPP${{ inputs.firebase_cpp_sdk_version }}
```
Comment (sunmou99) [inline .github/workflows/build_android.yml:40]: remove `macos` here?
`build-android-unity${{ inputs.unity_version }}-CPP${{ inputs.firebase_cpp_sdk_version }}`

File: .github/workflows/build_ios.yml [hunk 2]
```diff
@@ -8,30 +8,36 @@ on:
...
       additional_cmake_flags:
         description: 'Additional flags to pass into CMake'
         default: ''
         required: false
+        type: string
 
 jobs:
   build_ios:
-    name: build-ios-macos-${{ github.event.inputs.unity_version }}-CPP${{ github.event.inputs.firebase_cpp_sdk_version }}
+    name: build-ios-macos-unity${{ inputs.unity_version }}-CPP${{ inputs.firebase_cpp_sdk_version }}
```
Comment (sunmou99) [inline .github/workflows/build_ios.yml:40]: same here, remove `macos`?

Comment (cynthiajoan) [reply .github/workflows/build_ios.yml]: done

File: .github/workflows/build_android.yml (hunk 1 above)
Comment (cynthiajoan) [reply .github/workflows/build_android.yml]: done

File: .github/workflows/build_starter.yml [hunk 3]
```diff
@@ -28,87 +28,94 @@ on:
...
+    name: build-macos-unity${{ github.event.inputs.unity_version }}-CPP${{ github.event.inputs.firebase_cpp_sdk_version }}
+    uses: ./.github/workflows/build_macos.yml
+    needs: check_and_prepare
+    if: contains(needs.check_and_prepare.outputs.platform, 'macOS')
+    with:
+      unity_version: ${{ github.event.inputs.unity_version }}
+      firebase_cpp_sdk_version: ${{ github.event.inputs.firebase_cpp_sdk_version }}
+      unity_branch: ${{ github.event.inputs.unity_branch }}
+      apis: ${{ github.event.inputs.apis }}
+      unity_platform_name: macOS,iOS
```
Comment (sunmou99) [inline .github/workflows/build_starter.yml:107]: iOS?

Comment (cynthiajoan) [reply .github/workflows/build_starter.yml:107]: I'm mapping this from the existing sdk build process, which does need both of the component installed.

**Review by sunmou99 — APPROVED** [review]
LGTM

**Pull request merged**